### ISSUE TITLE
Result from REST API different from website

### ISSUE THREAD

**Issue by cambp4** [compiler-explorer/compiler-explorer#2579]
Hi, 

**Describe the bug**
I am developing a plugin and I realised that the result from the REST API is different that the one when I use the website

**To Reproduce**
this is the request sent via curl
curl -X POST -H "Content-Type: application/json" -d @json_main.json https://godbolt.org/api/compiler/g82/compile

this is the snippet of code as well as the assembly result from the REST API
![image](https://user-images.githubusercontent.com/62029155/113741037-2b280c00-96f9-11eb-9926-1bda33f0c685.png)

this is the options I send as part of the json file sent with the request:
             "demangle": true,
                "directives": true,
                "execute": false,
                "intel": true,
                "labels": true,
                "libraryCode": false,
                "trim": false
        },
        "tools": [
                 {"id":"clangtidytrunk", "args":"-checks=*"}
         ],
         "libraries": [
                 {"id": "fmt", "version": "trunk"}
         ]
    },
    "lang": "c++",
    "allowStoreCodeDebug": false,
    "instructionSet": "amd64"

**Expected behavior**
I would expect a similar assembly that the one here https://godbolt.org/z/E938c4TaW

**Comment by partouf** [comment]
In the JSON options you did not specify `-O3` (specifically `"userArguments": "-O3"`)

**Comment by partouf** [comment]
I'm also quite confused about the labels both internal and for fmt. Are you sure you're sending this to g82? The namespace for the fmt library functions should be `fmt::v7::`, not just `fmt::`

**Comment by cambp4** [comment]
sorry somehow the copy paste did not work as expected. 
this is the option file that I use to construct the json file.  I set library code to true but it does not seem to make a difference on the assembly (the assembly is still different from the one from the website)
   
![image](https://user-images.githubusercontent.com/62029155/113753130-a7285100-9705-11eb-96ab-bd32af8d7a27.png)

**Comment by partouf** [comment]
Can you try and see what's returned when you add `"bypassCache": true`?

(btw "instructionSet" is not part of the compile API)

**Comment by mattgodbolt** [comment]
By using the "Network" tab in a browser you can usually see the exact queries made. For example, the CE link you sent itself posts:

```json
{
  "source": "#include <string>\n\n#include <fmt/format.h>\n\nint main() {\n  const std::string s = fmt::format(\"The answer is {}.\", 42);\n  fmt::print(\"The answer is {}.\", 42);\n  return 0;\n}",
  "compiler": "g82",
  "options": {
    "userArguments": "-O3",
    "compilerOptions": {
      "produceGccDump": {},
      "produceCfg": false
    },
    "filters": {
      "binary": false,
      "execute": false,
      "intel": true,
      "demangle": true,
      "labels": true,
      "libraryCode": true,
      "directives": true,
      "commentOnly": true,
      "trim": false
    },
    "tools": [],
    "libraries": [
      {
        "id": "fmt",
        "version": "trunk"
      }
    ]
  },
  "lang": "c++",
  "allowStoreCodeDebug": true
}
```

As partouf says you can add `bypassCache` for debugging (but please don't use it in general, caching helps us).

If possible can you output your exact, text, JSON request, instead of a partial screenshot?

**Comment by cambp4** [comment]
hi, thanks both of you for your time

by copy pasting the json @mattgodbolt gave I managed to get the same result. also thanks for the tip with the network tab that something I learned today.

for info my json file was similar to my option file as I'm just using a bash script to add the source field to the end of the option file and I add the curly braces at the beginning and at the end. 

anyway it seems to be solved now so thank you again guys :)
